parcels/fieldfilebuffer.py

```diff
@@ -48,79 +43,55 @@ def close(self):
-                lon_subset = np.array(lon[0, self.indices["lat"], self.indices["lon"]])
-                lat_subset = np.array(lat[0, self.indices["lat"], self.indices["lon"]])
+        if self.gridindexingtype not in ["croco"]:
+            if len(lon.shape) == 3:  # some lon, lat have a time dimension 1
+                lon = lon[0, :, :]
+                lat = lat[0, :, :]
             elif len(lon.shape) == 4:  # some lon, lat have a time and depth dimension 1
-                lon_subset = np.array(lon[0, 0, self.indices["lat"], self.indices["lon"]])
-                lat_subset = np.array(lat[0, 0, self.indices["lat"], self.indices["lon"]])
-
-        if len(lon.shape) > 1:  # Tests if lon, lat are rectilinear but were stored in arrays
-            rectilinear = True
-            # test if all columns and rows are the same for lon and lat (in which case grid is rectilinear)
-            for xi in range(1, lon_subset.shape[0]):
-                if not np.allclose(lon_subset[0, :], lon_subset[xi, :]):
-                    rectilinear = False
-                    break
-            if rectilinear:
-                for yi in range(1, lat_subset.shape[1]):
-                    if not np.allclose(lat_subset[:, 0], lat_subset[:, yi]):
-                        rectilinear = False
-                        break
-            if rectilinear:
-                lon_subset = lon_subset[0, :]
-                lat_subset = lat_subset[:, 0]
-        return lat_subset, lon_subset
+                lon = lon[0, 0, :, :]
+                lat = lat[0, 0, :, :]
+
+        if len(lon.shape) > 1:
+            if is_rectilinear(lon, lat):
+                lon = lon[0, :]
+                lat = lat[:, 0]
+        return lat, lon
 
     @property
     def depth(self):
         if "depth" in self.dimensions:
             depth = self.dataset[self.dimensions["depth"]]
-            depthsize = depth.size if len(depth.shape) == 1 else depth.shape[-3]
-            if self.gridindexingtype in ["croco"]:
-                depthsize -= 1
-            self.data_full_zdim = depthsize
-            self.indices["depth"] = self.indices["depth"] if "depth" in self.indices else range(depthsize)
-            if len(depth.shape) == 1:
-                return np.array(depth[self.indices["depth"]])
-            elif len(depth.shape) == 3:
-                if self.nolonlatindices:
-                    return np.array(depth[self.indices["depth"], :, :])
-                else:
-                    return np.array(depth[self.indices["depth"], self.indices["lat"], self.indices["lon"]])
-            elif len(depth.shape) == 4:
-                if self.nolonlatindices:
-                    return np.array(depth[:, self.indices["depth"], :, :])
-                else:
-                    return np.array(depth[:, self.indices["depth"], self.indices["lat"], self.indices["lon"]])
+            self.data_full_zdim = self.zdim
```

Comment:
This difference between fulL_zdim and dim should also be removed at some point. This should be handled by array and the interpolation methods. Or don't we need to worry too much because at some point the whole fieldfilebuffer.py will be gone - as we use array for all this?

Reply:
> Or don't we need to worry too much because at some point the whole fieldfilebuffer.py will be gone - as we use array for all this?

Indeed I think we don't need to worry about this for now. Down the line I can see a lot of this transformation being lazy methods on top of xarray objects